From f84964ef0bd240dfe9ec822e31e1a5817fe34c85 Mon Sep 17 00:00:00 2001
From: Michael B. Trausch <mike@trausch.us>
Date: Fri, 29 May 2009 22:45:36 -0400
Subject: [PATCH] Fix Gdk.XEvent GDK binding

This commit partially reverts git commit bcdf1d1a3, restoring the
generation of Gdk.XEvent as a pointer.  This is in preparation for a
fix to the X11 bindings which is required for AllTray (which uses
those bindings) to build properly with Vala > 0.7.3.

Signed-off-by: Michael B. Trausch <mike@trausch.us>
---
 vapi/gdk-2.0.vapi                      |    7 ++++---
 vapi/packages/gdk-2.0/gdk-2.0.metadata |    1 -
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/vapi/gdk-2.0.vapi b/vapi/gdk-2.0.vapi
index 175fcc4..a2c4dd2 100644
--- a/vapi/gdk-2.0.vapi
+++ b/vapi/gdk-2.0.vapi
@@ -592,6 +592,10 @@ namespace Gdk {
 		public void unstick ();
 		public void withdraw ();
 	}
+	[Compact]
+	[CCode (cheader_filename = "gdk/gdk.h")]
+	public class XEvent {
+	}
 	[CCode (cheader_filename = "gdk/gdk.h")]
 	[SimpleType]
 	public struct Atom {
@@ -968,9 +972,6 @@ namespace Gdk {
 	public struct WindowRedirect {
 		public static void to_drawable (Gdk.Window window, Gdk.Drawable drawable, int src_x, int src_y, int dest_x, int dest_y, int width, int height);
 	}
-	[CCode (type_id = "GDK_TYPE_XEVENT", cheader_filename = "gdk/gdk.h")]
-	public struct XEvent {
-	}
 	[CCode (cprefix = "GDK_AXIS_", cheader_filename = "gdk/gdk.h")]
 	public enum AxisUse {
 		IGNORE,
diff --git a/vapi/packages/gdk-2.0/gdk-2.0.metadata b/vapi/packages/gdk-2.0/gdk-2.0.metadata
index 829ac5a..8bfc9c8 100644
--- a/vapi/packages/gdk-2.0/gdk-2.0.metadata
+++ b/vapi/packages/gdk-2.0/gdk-2.0.metadata
@@ -133,4 +133,3 @@ gdk_window_get_position.y is_out="1"
 gdk_window_get_root_origin.x is_out="1"
 gdk_window_get_root_origin.y is_out="1"
 gdk_window_invalidate_rect.rect nullable="1"
-GdkXEvent is_value_type="1"
-- 
1.6.0.4

